app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt

```diff
@@ -16,7 +18,8 @@ class MovieDetailsFragment : Fragment(R.layout.fragment_movie_details) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding = FragmentMovieDetailsBinding.bind(view)
-        requireArguments().getParcelable<Movie>(ARG_MOVIE)?.let {
+        requireArguments().getString(ARG_MOVIE)?.let { viewModel.setMovie(it) }
```

Comment:
setMovie? Is that really what this method do?

Reply:
fixed